lib/chef/policy_builder/expand_node_object.rb

```diff
@@ -54,6 +55,21 @@ def initialize(node_name, ohai_data, json_attribs, override_runlist, events)
         @run_list_expansion = nil
       end
 
+      # This method injects the run_context and provider and resource priority
```

Comment:
I want to be able to hit Chef.run_context from a cookbook library for other reasons (right now the run_context is effectively a singleton, but without an `instance` method, so if you want to hook it at class-loading time in cookbook library code you have to search ObjectSpace to find it).

Its sort of unrelated to this change, but related to setting up the Chef class as part of the DSL to access Chef internals.